### PR TITLE
WIP New post: Continuous Delivery Tools Revisited

### PR DESCRIPTION
**Pull request by booyaa**
Could I please get some feedback on this draft post. In particular, I have questions about: 
- If I've missed any examples of our use of tools with our clients (good or bad)
- Anything that our partners (AWS, RedHat, etc) offer in this space
- Any new features in this space that I haven’t already included

TODO
- [x] Finish Summary
- [ ] Technnical Review
- [x] Editorial Review

### CODE REVIEW

File: posts/continuous_delivery_tools_revisited.md [hunk 1]
```diff
@@ -0,0 +1,41 @@
+# Continuous Delivery Tools Revisited
+
+Since we wrote the original [article](https://www.madetech.com/blog/continuous-delivery-tools) just a little over two years ago, we've seen a fairly big shift away from self-hosted tools to feature rich Software as Service (SaaS). We've also seen nearly all Continuous Integration (CI) tools blur their lines with Continuous Delivery (CD) providing an all-in-one solution.
+
+We'll see who are the new players, what are the new features to look for, and then review the advice we gave regarding choosing self-hosted over SaaS.
```
Comment (EThom73) [inline posts/continuous_delivery_tools_revisited.md]: We'll see who the new players are,

File: posts/continuous_delivery_tools_revisited.md [hunk 2]
```diff
@@ -0,0 +1,41 @@
...
+
+## Self-hosted
+
+[GitLab](https://about.gitlab.com/pricing/#self-managed) aims to provide seamless integration between self-hosted and SaaS. With multiple ways to deploy your own CI/CD instance. The added bonus of using GitLab is that you can buy support.
+
+## SaaS
+
+There's been a massive growth in this area with cloud independent solutions such as [Travis CI](https://travis-ci.org/), [Circle CI](https://circleci.com/). At the same time, Microsoft's Azure Pipeline strong integration with their cloud offering.
```
Comment (EThom73) [inline posts/continuous_delivery_tools_revisited.md]: > At the same time, Microsoft's Azure Pipeline strong integration with their cloud offering.

missing a verb

File: posts/continuous_delivery_tools_revisited.md [hunk 3]
```diff
@@ -0,0 +1,41 @@
...
+
+## New Features
+
+### Docker support
+
+Container images have replaced buildpacks as the way to assure consistent versions of software releases. Look for baked in Docker support, to reduce the complexity and overhead of your pipeline.
```
Comment (EThom73) [inline posts/continuous_delivery_tools_revisited.md]: >Look for baked in Docker support, to reduce the complexity and overhead of your pipeline.

Comma removal: Look for baked in Docker support to reduce the complexity and overhead of your pipeline.

File: posts/continuous_delivery_tools_revisited.md [hunk 4]
```diff
@@ -0,0 +1,41 @@
...
+
+### Pipelines/Workflow
+
+This is the heart of continuous delivery and there has been a lot of innovation in this space.
+
+[GitHub](https://github.com/) who traditionally are the first step in building a CD pipeline now provides [GitHub Actions](https://developer.github.com/actions/) which allow pipeline functionality from within your source code repository. It provides a similar "look and feel" to [IFTTT](https://ifttt.com/) allowing build your workflow visually.
```
Comment (EThom73) [inline posts/continuous_delivery_tools_revisited.md]: > [GitHub](https://github.com/) who traditionally are the first step in building a CD pipeline now provides...

Comma usage: [GitHub](https://github.com/), who traditionally are the first step in building a CD pipeline, now provides

File: posts/continuous_delivery_tools_revisited.md [hunk 5]
```diff
@@ -0,0 +1,41 @@
...
+
+Circle CI have turned workflow steps into shareable packages called Orbs.
+
+The "deploy anywhere" approach of [Microsoft Azure Pipelines](https://azure.microsoft.com/en-us/services/devops/pipelines/) (part of the Azure DevOps suite) means you can release software to any environment Kubernetes (Azure Kubernetes Service),  Serverless (Azure Functions) and Web. This is, of course, achievable because of the close integration with their Cloud platform Azure.
```
Comment (EThom73) [inline posts/continuous_delivery_tools_revisited.md]: >software to any environment 

add colon. software to any environment :

File: posts/continuous_delivery_tools_revisited.md [hunk 6]
```diff
@@ -0,0 +1,41 @@
...
+
+## Self Hosted or SaaS
+
+The general advice from our original article remains the same, if it's a single product then SaaS will provide the least amount of friction whereas if you have multiple products or at enterprise scale then self-hosted is still worth considering.
```
Comment (EThom73) [inline posts/continuous_delivery_tools_revisited.md]: >The general advice from our original article remains the same, if it's a single product then SaaS will provide the least amount of friction whereas if you have multiple products or at enterprise scale then self-hosted is still worth considering.

couple of punctuation edits: The general advice from our original article remains the same. If it's a single product then SaaS will provide the least amount of friction, whereas if you have multiple products, or are at enterprise scale, then self-hosted is still worth considering.

File: posts/continuous_delivery_tools_revisited.md [hunk 7]
```diff
@@ -0,0 +1,41 @@
...
+
+In addition to this advice, we now also recommend that if you are delivering to multiple operating systems (Windows, MacOS and Linux) then SaaS can reduce the complexity of managing an exotic build farm. A common combination used by Rust developers is to use Travis CI for Linux and MacOS and [Appveyor](https://duckduckgo.com/?q=appveyor&t=ffab&ia=web) for Windows when building and packaging software.
+
+The self-hosted route is only feasible if you have a dedicated Ops team, as the chances are if your CD goes down, your CI will be down too (nearly all self-hosted solution provide CI/CD as an all-in-one package). If both are down then your developers can't release features.
```
Comment (EThom73) [inline posts/continuous_delivery_tools_revisited.md]: edit: as the chances are **that** if your CD goes down

**Review by EThom73 — CHANGES_REQUESTED** [review]
Left some reviews on there. Only a couple of readability/punctuation things.

File: posts/continuous_delivery_tools_revisited.md [hunk 8]
```diff
@@ -0,0 +1,49 @@
...
+
+[GitLab](https://about.gitlab.com/pricing/#self-managed) aims to provide seamless integration between self-hosted and SaaS. With multiple ways to deploy your own CI/CD instance. The added bonus of using GitLab is that you can buy support.
+
+*Note: Block quotes are temporary and will be removed before publication*
+
+> We have used this package microservices as charts (a deployable unit in Helm, which is Kubernetes' package manager) to be deployed to the Kubernetes cluster. And yes we hosted our GitLab in our cluster.
+
+## SaaS
+
+There's been a massive growth in this area with cloud independent solutions such as [Travis CI](https://travis-ci.org/), [Circle CI](https://circleci.com/). On the flip side Microsoft's Azure Pipeline has strong integration with their cloud offering Azure.
```
Comment (adrianclay) [inline posts/continuous_delivery_tools_revisited.md]: There's [AWS CodeBuild](https://aws.amazon.com/codebuild/) too.

**Review by EThom73 — APPROVED** [review]

File: posts/continuous_delivery_tools_revisited.md (hunk 8 above)
Comment (SebAshton) [reply posts/continuous_delivery_tools_revisited.md]: If we're doing CD pipelines do we need to mention [AWS CodeDeploy](https://aws.amazon.com/codedeploy/) too

Comment (SebAshton) [reply posts/continuous_delivery_tools_revisited.md]: Heroku do CI now too

File: posts/continuous_delivery_tools_revisited.md [hunk 9]
```diff
@@ -0,0 +1,49 @@
...
+
+## New Features
+
+### Docker support
+
+Container images have replaced buildpacks as the way to assure consistent versions of software releases. Look for baked in Docker support to reduce the complexity and overhead of your pipeline.
+
+> We use docker images as a consistent way to base image for Ruby...
```
Comment (SebAshton) [inline posts/continuous_delivery_tools_revisited.md]: Could link to our Docker Hub

File: posts/continuous_delivery_tools_revisited.md [hunk 10]
```diff
@@ -0,0 +1,49 @@
...
+
+### Pipelines/Workflow
+
+This is the heart of continuous delivery and there has been a lot of innovation in this space.
+
+[GitHub](https://github.com/), who traditionally are the first step in building a CD pipeline now provides [GitHub Actions](https://developer.github.com/actions/) which allow pipeline functionality from within your source code repository. It provides a similar "look and feel" to [IFTTT](https://ifttt.com/) allowing build your workflow visually.
+
+Circle CI have turned workflow steps into shareable packages called Orbs.
```
Comment (SebAshton) [inline posts/continuous_delivery_tools_revisited.md]: Heroku has a deployment pipeline and offers review apps.

File: posts/continuous_delivery_tools_revisited.md (hunk 9 above)
Comment (SebAshton) [reply posts/continuous_delivery_tools_revisited.md]: https://hub.docker.com/u/madetech/

File: posts/continuous_delivery_tools_revisited.md [hunk 11]
```diff
@@ -0,0 +1,59 @@
+# Continuous Delivery Tools Revisited
+
+Since we wrote the original [article](https://www.madetech.com/blog/continuous-delivery-tools) just a little over two years ago, we've seen a fairly big shift away from self-hosted tools to feature rich Software as Service (SaaS). We've also seen nearly all Continuous Integration (CI) tools blur their lines with Continuous Delivery (CD) providing an all-in-one solution.
+
+We'll see who the new players are, what are the new features to look for, and then review the advice we gave regarding choosing self-hosted over SaaS.
+
```
Comment (DelphineGattaz) [inline posts/continuous_delivery_tools_revisited.md:6]: I think we're missing a sentence here to explain why we're talking about Self-hosted or Saas. What is the angle for the next two sections? are we comparing?

**Pull request closed without merging**